Tests/ApolloCodegenTests/GlobTests.swift

```diff
@@ -0,0 +1,359 @@
+    expect(Glob([pattern]).match).to(equal([
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path,
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.two").path
+    ]))
+  }
+
+  func test_match_givenGlobstarPattern_usingSingleWildcard_whenSingleMatch_shouldReturnSingle() throws {
+    // given
+    let pattern = baseURL.appendingPathComponent("a/b/c/d/**/?ile.one").path
+
+    // when
+    // <baseURL>/a/b/c/d/e/f/file.one
+    // <baseURL>/a/b/c/d/e/f/file.two
+
+    // then
+    expect(Glob([pattern]).match).to(equal([
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path
+    ]))
+  }
+
+  func test_match_givenGlobstarPattern_usingCombinedWildcard_whenMultipleMatch_shouldReturnMultiple() throws {
+    // given
+    let pattern = baseURL.appendingPathComponent("a/b/c/d/**/fil?.*").path
+
+    // when
+    // <baseURL>/a/b/c/d/e/f/file.one
+    // <baseURL>/a/b/c/d/e/f/file.two
+
+    // then
+    expect(Glob([pattern]).match).to(equal([
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path,
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.two").path
+    ]))
+  }
+
+  func test_match_givenPattern_withExcludeNotFirst_shouldThrow() throws {
+    // given
+    let pattern = baseURL.appendingPathComponent("a/b/c/d/**/!file.swift").path
+
+    // then
+    expect(Glob([pattern]).match).to(throwError(Glob.MatchError.invalidExclude(path: pattern)))
+  }
+
+  func test_match_givenGlobstarPattern_usingPathExclude_whenMultipleMatch_shouldExclude() throws {
+    // given
+    let pattern = [
+      baseURL.appendingPathComponent("a/b/c/d/**/file.*").path,
+      "!" + baseURL.appendingPathComponent("a/b/c/d/**/file.two").path,
+    ]
+
+    // when
+    // <baseURL>/a/b/c/d/e/f/file.one
+    // <baseURL>/a/b/c/d/e/f/file.two
+
+    // then
+    expect(Glob(pattern).match).to(equal([
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path
+    ]))
+  }
+
+  func test_match_givenRelativePattern_usingNoPrefix_shouldUseCurrentDirectory() throws {
+    // given
+    let pattern = ["**/*.one"]
+
+    FileManager.default.changeCurrentDirectoryPath(baseURL.path)
+
+    // when
+    // <baseURL>/file.one
+    // <baseURL>/file.two
+    // <baseURL>/a/file.one
+    // <baseURL>/a/b/file.one
+    // <baseURL>/a/b/c/file.one
+    // <baseURL>/a/b/c/d/e/f/file.one
+    // <baseURL>/a/b/c/d/e/f/file.two
+    // <baseURL>/other/file.one
+    // <baseURL>/other/file.oye
+
+    // then
+    expect(Glob(pattern).match).to(equal([
+      baseURL.appendingPathComponent("file.one").path,
+      baseURL.appendingPathComponent("a/file.one").path,
+      baseURL.appendingPathComponent("a/b/file.one").path,
+      baseURL.appendingPathComponent("a/b/c/file.one").path,
+      baseURL.appendingPathComponent("a/b/c/d/e/f/file.one").path,
+      baseURL.appendingPathComponent("other/file.one").path
+    ]))
+  }
+
+  func test_match_givenRelativePattern_usingSingleDotPrefix_shouldUseCurrentDirectory() throws {
```

Comment:
Probably worth also testing that if the current directory is the `a` directory, you're not getting `file.one` or `other/file.one` for this same search

Reply:
Good call - a test for this was added in https://github.com/apollographql/apollo-ios/pull/2041/commits/3d0d633363ef6104b7ea28a0f1925cfa80e6cb7d.